spec/services/service_provider_logo_updater_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe ServiceProviderLogoUpdater do
+  include ServiceProviderHelper
+
+  subject(:updater) { ServiceProviderLogoUpdater.new }
+
+  let(:service_provider) { create(:service_provider, :with_team, logo: 'logo.svg') }
+  let(:fake_config) { [service_provider.attributes] }
+  let(:fake_response) do
+    OpenStruct.new(
+      body: JSON.generate(fake_config)
+    )
+  end
+
+  describe '#import_logos_to_active_storage' do
+    it 'attaches the legacy logo to the service provider' do
+      # don't clone the config repo
+      allow(Subprocess).to receive(:check_call).and_return(true)
+      # don't call /api/service_providers on the dashboard
+      # allow(updater).to receive(:load_idp_config).and_return(fake_config)
+      allow_any_instance_of(RestClient::Request).to receive(:execute).and_return(fake_response)
```

Comment:
we should be able to stub this with webmock's `stub_request` instead of having to do `allow_any_instance_of` (ditto for the rest of this file too)